File: src/main/java/dev/vankka/simpleast/core/node/TextNode.java
package dev.vankka.simpleast.core.node;

import lombok.RequiredArgsConstructor;

/**
 * Node representing simple text.
 */
@RequiredArgsConstructor
public class TextNode<R> extends Node<R> {
    private final String content;

    public String getContent() {
        return content;
    }

    @Override
    public String toString() {
        return getClass().getSimpleName() + "{content=" + content + ", children=" + getChildren() + "}";
    }
}

File: src/main/java/dev/vankka/simpleast/core/node/TextNode.kt
package dev.vankka.simpleast.core.node

import java.lang.StringBuilder

/**
 * Node representing simple text.
 */
open class TextNode<R> (val content: String) : Node<R>() {
  override fun render(builder: StringBuilder, renderContext: R) {
    builder.append(content)
  }

  override fun toString() = "${javaClass.simpleName}[${getChildren()?.size}]: $content"
}
